snakebite/client.py

```diff
@@ -1425,10 +1476,16 @@ def __handle_socket_error(self, exception):
         log.debug("Request failed with %s" % exception)
         if exception.errno in (errno.ECONNREFUSED, errno.EHOSTUNREACH):
             # if NN is down or machine is not available, pass it:
-            pass
+            if self.__should_retry():
```

Comment:
Does it always make sense to retry here? Might this sometimes delay "failover" to maybe active namenode?